src/imageproxy.rs

```diff
@@ -166,15 +171,24 @@ impl TryFrom<ImageProxyConfig> for Command {
             c
         });
         c.arg("experimental-image-proxy");
-        if let Some(authfile) = config.authfile.as_deref() {
-            c.args(&["--authfile", authfile]);
+        if let Some(authfile) = config.authfile {
             if config.auth_anonymous {
                 // This is a programmer error really
                 anyhow::bail!("Cannot use anonymous auth and an authfile");
             }
+            match authfile.to_str() {
+                Some(authfile) => c.args(&["--authfile", authfile]),
+                None => anyhow::bail!("Failed to decode path into string"),
+            };
```

Comment:
So this is an interesting topic.  What you have here is I'd say fine, because I certainly don't care about supporting non-UTF-8 paths.

However, there's also not a reason to break on it.  It's simpler to do this:

```suggestion
            c.arg("--authfile");
            c.arg(authfile);
```

What you hit is that in Rust, an array/vector must have all elements of the same type, and a `PathBuf` is a distinct thing from a `&str`.

The [.arg() method](https://doc.rust-lang.org/std/process/struct.Command.html#method.arg) accepts a single value, but it can be any type that can be converted into an `OsStr` at each call site.  So using separate calls to `.arg()` should work.

This way, if someone passes us a non-UTF-8 path, we will just send it through to the child process.  Where, it will probably work, although this is rarely tested of course.  Anyways, it's less code.

Let's do the same with the certificate bits.